library/src/blas3/Tensile/gemm_strided_batched.cpp

```diff
@@ -98,13 +98,13 @@ namespace
                 if(layer_mode & rocblas_layer_mode_log_bench)
                 {
                     std::stringstream alphass;
-                    alphass << "--alpha " << std::real(*alpha);
-                    if (std::imag(*alpha) != 0)
+                    alphass << "--alpha " << (alpha ? std::real(*alpha) : std::numeric_limits<T>::quiet_NaN());
+                    if (alpha && std::imag(*alpha) != 0)
                         alphass << " --alphai " << std::imag(*alpha);
 
                     std::stringstream betass;
-                    betass << "--beta " << std::real(*beta);
-                    if (std::imag(*beta) != 0)
+                    betass << "--beta " << (beta ? std::real(*beta) : std::numeric_limits<T>::quiet_NaN());
+                    if (beta && std::imag(*beta) != 0)
                         betass << " --betai " << std::imag(*beta);
 
                     log_bench(handle,
```

Comment:
Also wondered if we could make some argument class for blas1, blas2, blas3 to indicate what standard arguments they take.   Then a helper class with static methods could maybe do all the logging, layer, … work for all functions.     Don't know if we care about minor efficiency changes for all these methods but it would keep things consistent and reduce code.

Reply:
Every routine is separate, and there is not a "standard argument list" for BLAS1, BLAS2, or BLAS3. With very few exceptions in BLAS1, the argument list is different for every routine.

If we merged several routines' argument lists into a universal struct like `Arguments`, and printed that, then we would be printing a lot of extraneous arguments.

That's why the logging system is flexible, allowing arbitrary argument lists to be specified for each routine. The logging functions accept an arbitrary number of arguments.

It took around 2 months to implement the current logging system. With other priorities, we don't have the time to rewrite the logging system.

Related to that, the current `Arguments` struct is _universal_ for all of rocBLAS, and splitting it into separate classes for BLAS1, BLAS2, BLAS3, etc. would require **extensive** modifications across multiple languages (C++, Python, YAML).